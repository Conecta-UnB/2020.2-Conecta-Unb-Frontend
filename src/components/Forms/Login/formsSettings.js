import * as Yup from 'yup';

export const initialValues = {
  email: '',
  senha: '',
};

export const SignupSchema = Yup.object().shape({
  email: Yup
    .string()
    .email('Email inváilido')
    .required('É necessário indicar um email'),
  senha: Yup
    .string()
    .min(6, 'A senha deve ter pelo menos 6 dígitos')
    .required('É necessário criar uma senha'),

});

export const handleSubmit = async (values, setSubmitting, login) => {
  setSubmitting(false);
  await login({
    email: values.email,
    senha: values.senha,
  });
};
