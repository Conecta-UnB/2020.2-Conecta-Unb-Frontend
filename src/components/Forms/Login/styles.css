.containerLoginForm{
  background-color: #0D8447;
  width: 400px;
  height: 450px;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.imgContainer{
  display: flex;
  width: 200px;
  padding-right: 8%;
  height: 200px;
  justify-content: center;
}

.imgContainer img {
  border-radius: 50%;
}

.containerLoginForm div{
  align-self: center;
  padding-top: 20px;
}

.containerLoginForm input{
  width: 350px;
  margin-left: 20px;
  margin-right: 20px;
  height: 20px;
  border-radius: 20px;
  margin-top: 10px;
  margin-bottom: 20px;
  padding: 20px;
}

.containerLoginForm label{
  font-size: 20px;
  margin-left: 20px;
  color: #ffff;
  font-weight: bold;
  padding-top: 60px;
  margin-bottom: 40px;
}

.containerLoginForm button{
  background: #0260EE;
  align-self: center;
  display: block;
  margin-top: 10px;
  margin-left: auto;
  margin-right: auto;
  width: 60px;
  height: 30px;
  border-radius: 8px;
  border: none;
  color: #ffff;
  font-weight: bold;
}