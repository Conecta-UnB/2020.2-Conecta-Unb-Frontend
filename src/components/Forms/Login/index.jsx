/* eslint-disable react/button-has-type */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import logo from '../../../assets/logo.jpeg';
import './styles.css';

export default function LoginForm() {
  return (
    <div className="containerLoginForm">
      <div>
        <img src={logo} alt="logoImage" />
        <form>
          <label>
            E-mail:
          </label>
          <input type="email" autoComplete="off" />
          <label>
            Senha:
          </label>
          <input type="password" autoComplete="off" />
          <button type="submit">
            Entrar
          </button>
        </form>
      </div>
    </div>
  );
}
