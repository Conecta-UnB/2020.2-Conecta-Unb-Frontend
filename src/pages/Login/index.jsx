import React from 'react';
import LoginForm from '../../components/Forms/Login';
import './styles.css';

export default function LoginPage() {
  return (
    <div className="underHeader">
      <LoginForm />
    </div>
  );
}
