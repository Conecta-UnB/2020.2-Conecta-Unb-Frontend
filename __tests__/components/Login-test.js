import React from 'react';
import { render } from '@testing-library/react';
import Login from '../../src/components/Forms/Login';

describe('testando loginforms', () => {
  test('deve ser igual a snapshot', () => {
    const { asFragment } = render(<Login />);
    expect(asFragment(<Login />)).toMatchSnapshot();
  });
});
