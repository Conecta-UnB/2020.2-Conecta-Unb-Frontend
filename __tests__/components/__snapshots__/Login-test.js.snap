// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando loginforms deve ser igual a snapshot 1`] = `
<DocumentFragment>
  <div
    class="containerLoginForm"
  >
    <div
      class="imgContainer"
    >
      <img
        alt="logoImage"
        src="[object Object]"
      />
    </div>
    <form>
      <label>
        E-mail:
      </label>
      <input
        autocomplete="off"
        type="email"
      />
      <label>
        Senha:
      </label>
      <input
        autocomplete="off"
        type="password"
      />
      <button
        type="submit"
      >
        Entrar
      </button>
    </form>
  </div>
</DocumentFragment>
`;
